driver-core/src/main/com/mongodb/operation/CommandOperationHelper.java

```diff
@@ -895,6 +895,10 @@ static boolean isRetryableException(final Throwable t) {
             return true;
         }
         String errorMessage = t.getMessage();
+        if (((MongoException) t).getCode() == 20 && errorMessage.contains("Transaction numbers")) {
```

Comment:
Change to

    errorMessage.startWith...

as per spec

Reply:
The string returned from `getMessage()` is:
 `Command failed with error 20 (IllegalOperation): 'Transaction numbers are only allowed on storage engines that support document-level locking' on server localhost:27017. The full response is {"operationTime": {"$timestamp": {"t": 1565199509, "i": 1}}, "ok": 0.0, "errmsg": "Transaction numbers are only allowed on storage engines that support document-level locking", "code": 20, "codeName": "IllegalOperation", "$clusterTime": {"clusterTime": {"$timestamp": {"t": 1565199509, "i": 1}}, "signature": {"hash": {"$binary": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=", "$type": "00"}, "keyId": {"$numberLong": "0"}}}}`
I don't see a way to get the error message as in the spec.

Reply:
oh, because the driver concatenates the server error message with that preamble.  ok.